test/statement_types.py

```diff
@@ -0,0 +1,451 @@
+        raise Exception("CppStatement should never be instantiated directly")
+
+    def is_reverse_mode(self):
+        """By default, a statement is not reverse mode"""
+        return False
+    
+    def is_matrix_like(self):
+        """By default, a statement is not matrix-like"""
+        return False
+
+    def is_varmat_compatible(self):
+        """By default, a statement is not compatible with varmat types"""
+        return False
+
+    def is_expression(self):
+        """By default, a statement is not an expression"""
+        return False
+
+class IntVariable(CppStatement):
+    """Represents an integer variable"""
+    def __init__(self, name, value = None):
+        """
+        Initialize matrix
+        
+        :param overload: Type of overload as string (Prim/Fwd/Rev/etc.)
+        :param name: C++ name of variable
+        :param stan_arg: Stanc3 type string
+        :param value: Scalar value to initialize matrix with
+        """
+        self.name = name
+        if value is None:
+            self.value = 1
+        else:
+            self.value = int(value)
+
+    def cpp(self):
+        """Generate C++"""
+        return "int {name} = {value};".format(name = self.name, value = self.value)
+
+class RealVariable(CppStatement):
+    """Represents a scalar, real variable"""
+    def __init__(self, overload, name, value = None):
+        """
+        Initialize matrix
+        
+        :param overload: Type of overload as string (Prim/Fwd/Rev/etc.)
+        :param name: C++ name of variable
+        :param stan_arg: Stanc3 type string
+        :param value: Scalar value to initialize matrix with
+        """
+        self.overload = overload
+        self.name = name
+        if value is None:
+            self.value = 0.4
+        else:
+            self.value = value
+    
+    def is_reverse_mode(self):
+        """Return true if the overload is reverse mode"""
+        return self.overload.startswith("Rev")
+    
+    def cpp(self):
+        """Generate c++"""
+        scalar = overload_scalar[self.overload]
+
+        return "{type} {name} = {value};".format(type = scalar, name = self.name, value = self.value)
+
+class MatrixVariable(CppStatement):
+    """Represents a matrix variable"""
+    def __init__(self, overload, name, stan_arg, value = None, size = None):
+        """
+        Initialize matrix
+        
+        :param overload: Type of overload as string (Prim/Fwd/Rev/etc.)
+        :param name: C++ name of variable
+        :param stan_arg: Stanc3 type string
+        :param value: Scalar value to initialize matrix with
+        :param size: Number of rows/columns in matrix
+        """
+        self.overload = overload
+        self.name = name
+        self.stan_arg = stan_arg
+        if value is None:
+            self.value = 0.4
+        else:
+            self.value = value
+        
+        if size is None:
+            self.size = 1
+        else:
+            self.size = size
```

Comment:
Set defaults to the values you want instead of this.

Reply:
I got rid of a lot of the default size behavior. I tried cleaning up the value defaults but realized I am using `value = None` to mean use a default value.